src/input.rs

```diff
@@ -232,3 +200,69 @@ where
         self.set_common(db, key, database_key, value, IsConstant(true))
     }
 }
+
+// Unsafe proof obligation: `Slot<DB, Q>` is Send + Sync if the query
+// key/value is Send + Sync (also, that we introduce no
+// references). These are tested by the `check_send_sync` and
+// `check_static` helpers below.
+unsafe impl<DB, Q> DatabaseSlot<DB> for Slot<DB, Q>
+where
+    Q: Query<DB>,
+    DB: Database,
+{
+    fn maybe_changed_since(&self, _db: &DB, revision: Revision) -> bool {
+        debug!(
+            "maybe_changed_since(slot={:?}, revision={:?})",
+            self, revision,
+        );
+
+        let changed_at = self.stamped_value.read().changed_at;
+
+        debug!("maybe_changed_since: changed_at = {:?}", changed_at);
+
+        changed_at.changed_since(revision)
+    }
+}
+
+/// Check that `Slot<DB, Q, MP>: Send + Sync` as long as
+/// `DB::DatabaseData: Send + Sync`, which in turn implies that
+/// `Q::Key: Send + Sync`, `Q::Value: Send + Sync`.
+#[allow(dead_code)]
```

Comment:
I tend to use `#[test]` for such checks, seems neat, but not necessary better

Reply:
I wanted it to be checked on every build, but yeah...

Reply:
`fn _check_static` is another option